float64 x
float64 y
float64 z
---
float64 theta1
float64 theta2
float64 d3
